cpp/include/cudf/detail/copy_if_else.cuh

```diff
@@ -44,18 +44,19 @@ __launch_bounds__(block_size) __global__
                            mutable_column_device_view out,
                            size_type* __restrict__ const valid_count)
 {
-  size_type const tid            = threadIdx.x + blockIdx.x * block_size;
-  int const warp_id              = tid / warp_size;
+  auto tidx                      = cudf::detail::grid_1d::global_thread_id();
+  auto const stride              = cudf::detail::grid_1d::grid_stride();
+  int const warp_id              = tidx / warp_size;
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**💡 suggestion:** ‏ Use `cudf::detail::warp_size`. CUDA's built-in `warp_size` is not a compile-time constant. (The reason is forever a mystery to me). Therefore we created a `constexpr` in `detail/utilities/cuda.cuh`. Could save multiple divisions here. Not sure how this slipped by in the past -- the constexpr is used in most cudf kernels.

Reply:
> Could save multiple divisions here. 

Sorry, I'm not sure I followed how multiple divisions might be saved here.  `tidx` isn't a `constexpr`, is it?

Reply:
If the compiler knows that warp_size is 32, it will use a SHR (right shift) instead of a DIV.

Reply:
Wow, that's brilliant.  Noted.

Reply:
Should have said this is the case for any power of 2.

Reply:
Done in 9e4154dd176c677ad7168c9f0770a7e2363c6970.